includes/class-amp-story-post-type.php

```diff
@@ -1368,4 +1371,48 @@ public static function remove_title_from_embed( $output, $post ) {
 
 		return preg_replace( '/<blockquote class="wp-embedded-content">.*?<\/blockquote>/', '', $output );
 	}
+
+	/**
+	 * Changes attributes of the AMP Story embed <iframe>.
+	 *
+	 * The embed typically appears in the editor in an <iframe>, and without an <iframe> on the front-end.
+	 * The height of the <iframe> isn't enough to display the full story, so this increases it.
+	 * Also, this removes attributes that AMP does not allow for the <amp-iframe>, like marginwidth.
+	 *
+	 * @param string  $output The embed output.
+	 * @param WP_Post $post The post for the embed.
+	 * @return string The filtered embed output.
+	 */
+	public static function change_embed_iframe_attributes( $output, $post ) {
+		if ( self::POST_TYPE_SLUG !== get_post_type( $post ) ) {
+			return $output;
+		}
+
+		preg_match( '/<iframe sandbox="allow-scripts" [^>]*>/', $output, $matches );
+		if ( ! $matches ) {
+			return $output;
+		}
+
+		$original_iframe = $matches[0];
+		$new_iframe      = preg_replace(
+			'/(?<=\sheight=")\w+(?=")/',
+			strval( ( self::STORY_LARGE_IMAGE_DIMENSION / 2 ) + 4 ),
```

Comment:
What is this magic number `4`?

Reply:
Hi @swissspidy, good to see you a few days ago.

The `<iframe>` apparently needs a little more height to display the entire embed `<a>`. 

Without the extra `4px`, here's what it looks like:

<img width="996" alt="without-extra-4" src="https://user-images.githubusercontent.com/4063887/57575788-106a9100-7417-11e9-8b99-2c7ff03267b1.png">

<img width="1311" alt="difference-in-iframe-height" src="https://user-images.githubusercontent.com/4063887/57575789-17919f00-7417-11e9-8b21-f971cf99997a.png">

Reply:
Gotcha, thanks. But why exactly `4`? Sounds a bit dangerous to rely on such a magic number.

Reply:
Yeah, this calculation definitely isn't ideal. It'd be nice to avoid this completely.

But it looks like `4px` is the increase in `<iframe>` height needed to ensure the image isn't cut off, and the `border-radius` appears:

# Adding 3px

<img width="1339" alt="adding-3px" src="https://user-images.githubusercontent.com/4063887/57576615-87a92080-7429-11e9-99b9-01f4c0731190.png">

# Adding 4px

<img width="1373" alt="adding-4px" src="https://user-images.githubusercontent.com/4063887/57576616-8e379800-7429-11e9-8e21-4e549a0cf3c4.png">